libmcount/dynamic.c

```diff
@@ -242,9 +242,36 @@ static int do_dynamic_update(struct symtabs *symtabs, char *patch_funcs,
 	strv_for_each(&funcs, name, j) {
 		bool found = false;
 		bool csu_skip;
+		char *modname, *delim;
 		unsigned i, k;
 		struct sym *sym;
 		struct uftrace_pattern patt;
+		struct uftrace_mmap *map;
+		struct mcount_dynamic_info *mdi = NULL;
+
+		delim = strchr(name, '@');
+
+		if (delim == NULL) {
+			// first of uftrace_mmap is main module always.
+			map = symtabs->maps;
+			symtab = &symtabs->maps->mod->symtab;
+		}
+		else {
+			*delim = '\0';
+			modname = ++delim;
+			map = find_map_by_name(symtabs, modname);
+			if (map == NULL) {
+				pr_dbg("Failed to find map of %s\n", modname);
+				continue;
+			}
+			symtab = &map->mod->symtab;
```

Comment:
You need to check if `map->mod` is NULL.

Reply:
fixed it!